src/main/java/io/phasetwo/service/auth/idp/HomeIdpDiscoverer.java

```diff
@@ -182,18 +182,13 @@ private List<IdentityProviderModel> discoverHomeIdps(Domain domain, UserModel us
             orgs.getUserOrganizationsStream(
                     context.getRealm(), user)
                 .filter(o -> {
-                    if(clientID.equals("sigsci-dashboard")) {
+                    String cid = o.getFirstAttribute("customer_id");
+                    boolean hasCID = cid != null && !cid.isEmpty();
+                    if(clientID.equals("sigsci-dashboard") || (!hasCID && clientID.contains("zendesk"))) {
```

Comment:
This has the problem that any corp IDP will take precedence over a fastly IDP when the client is zendesk.  I think we need to improve this condition.